spec/services/record_declarations/started/early_career_teacher_spec.rb

```diff
@@ -2,51 +2,290 @@
-
 RSpec.describe RecordDeclarations::Started::EarlyCareerTeacher do
-  include_context "lead provider profiles and courses"
-  include_context "service record declaration params"
+  let(:declaration_date_object) { Date.new(2021, 10, 1) }
+  let(:declaration_date) { declaration_date_object.rfc3339 }
+
+  let(:user) { profile.user }
+  let(:profile) { create(:ect_participant_profile) }
+
+  let(:cpd_lead_provider) { create(:cpd_lead_provider, :with_lead_provider) }
+  let(:lead_provider) { cpd_lead_provider.lead_provider }
+
+  let(:school_cohort) { create(:school_cohort) }
+  let(:school) { school_cohort.school }
+  let(:cohort) { school_cohort.cohort }
+
+  let(:partnership) { create(:partnership, school:, lead_provider:, cohort:) }
+  let(:induction_programme) { create(:induction_programme, :fip, partnership:) }
+
+  let(:participant_id) { profile.user.id }
+  let(:course_identifier) { "ecf-induction" }
+  let(:declaration_type) { "started" }
+
+  subject do
+    described_class.new(
+      params: {
+        participant_id:,
+        course_identifier:,
+        cpd_lead_provider:,
+        declaration_date:,
+        declaration_type:,
+      },
+    )
+  end
+
+  context "when evidence_held is provided" do
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+    end
+
+    subject do
+      described_class.new(
+        params: {
+          participant_id:,
+          course_identifier:,
+          cpd_lead_provider:,
+          declaration_date:,
+          declaration_type:,
+          evidence_held: "foo",
+        },
+      )
+    end
+
+    it "ignores the superfluous parameter" do
+      expect { subject.call }.to change { ParticipantDeclaration.count }.by(1)
+
+      new_declaration = ParticipantDeclaration.order(created_at: :asc).last
+
+      expect(new_declaration.evidence_held).to be_nil
+    end
+  end
+
+  context "when another lead provider makes the call" do
+    let(:other_cpd_lead_provider) { create(:cpd_lead_provider, :with_lead_provider) }
+    let(:other_lead_provider) { other_cpd_lead_provider.lead_provider }
+
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+    end
+
+    subject do
+      described_class.new(
+        params: {
+          participant_id:,
+          course_identifier:,
+          cpd_lead_provider: other_cpd_lead_provider,
+          declaration_date:,
+          declaration_type:,
+        },
+      )
+    end
+
+    it "raises an error" do
+      expect { subject.call }.to raise_error(ActionController::ParameterMissing, /The property '#\/participant_id' must be a valid Participant ID/)
+    end
+  end
+
+  context "happy path" do
+    before do
+      Induction::Enrol.new(induction_programme:, participant_profile: profile).call
+    end
+
+    it "creates the declaration" do
+      expect { subject.call }.to change { ParticipantDeclaration.count }.by(1)
```

Comment:
same a above, be a bit more specific in the query.

Reply:
done